recipes-kernel/linux/linux-yocto-onl-linux-5.15.y/bisdn-kmeta/bsp/x86_64-intel/x86_64-intel.cfg

```diff
@@ -2171,7 +2162,7 @@ CONFIG_I2C_ALGOBIT=y
 # CONFIG_I2C_AMD756 is not set
 # CONFIG_I2C_AMD8111 is not set
 # CONFIG_I2C_AMD_MP2 is not set
-CONFIG_I2C_I801=y
+CONFIG_I2C_I801=m
```

Comment:
Unfortunately this change breaks agema-ag7648:

```
delta-ag7648:~$ sudo onlpdump >/dev/null
11-03 12:07:55.890825 [onlplib] i2c-3: writing address 0x2a, offset 0 failed: No such device or address
11-03 12:07:55.891418 [onlplib] i2c-3: writing address 0x29, offset 0 failed: No such device or address
11-03 12:07:55.891966 [onlplib] i2c-3: writing address 0x2a, offset 1 failed: No such device or address
11-03 12:07:55.892504 [onlplib] i2c-3: writing address 0x29, offset 1 failed: No such device or address
11-03 12:07:55.893063 [onlplib] i2c-3: writing address 0x29, offset 2 failed: No such device or address
11-03 12:07:55.893628 [onlplib] i2c-3: writing address 0x29, offset 32 failed: No such device or address
11-03 12:07:55.894175 [onlplib] i2c-3: writing address 0x29, offset 33 failed: No such device or address
11-03 12:07:55.894735 [onlplib] i2c-3: writing address 0x29, offset 3 failed: No such device or address
11-03 12:07:55.895282 [onlplib] i2c-3: writing address 0x29, offset 34 failed: No such device or address
11-03 12:07:55.895835 [onlplib] i2c-3: writing address 0x29, offset 35 failed: No such device or address
11-03 12:07:55.896414 [onlplib] i2c-3: writing address 0x29, offset 4 failed: No such device or address
11-03 12:07:55.896954 [onlplib] i2c-3: writing address 0x29, offset 36 failed: No such device or address
11-03 12:07:55.897499 [onlplib] i2c-3: writing address 0x29, offset 37 failed: No such device or address
11-03 12:07:55.898019 [onlplib] i2c-3: writing address 0x29, offset 5 failed: No such device or address
11-03 12:07:55.898537 [onlplib] i2c-3: writing address 0x29, offset 38 failed: No such device or address
11-03 12:07:55.899058 [onlplib] i2c-3: writing address 0x29, offset 39 failed: No such device or address
11-03 12:07:55.899577 [onlplib] i2c-3: writing address 0x2a, offset 2 failed: No such device or address
11-03 12:07:55.900088 [onlplib] i2c-3: writing address 0x2a, offset 32 failed: No such device or address
11-03 12:07:55.900607 [onlplib] i2c-3: writing address 0x2a, offset 33 failed: No such device or address
11-03 12:07:55.901127 [onlplib] i2c-3: writing address 0x2a, offset 3 failed: No such device or address
11-03 12:07:55.901647 [onlplib] i2c-3: writing address 0x2a, offset 34 failed: No such device or address
11-03 12:07:55.902168 [onlplib] i2c-3: writing address 0x2a, offset 35 failed: No such device or address
11-03 12:07:55.928541 [onlplib] ONIE data contains an unrecognized TLV code: 0x1e (ignored)
11-03 12:07:55.935162 [onlplib] i2c-2: reading address 0x31, offset 0 failed: No such device or address
11-03 12:07:55.943341 [onlplib] i2c-2: reading address 0x32, offset 3 failed: No such device or address
11-03 12:07:55.950032 [onlplib] i2c-2: reading address 0x32, offset 3 failed: No such device or address
11-03 12:07:55.956723 [onlplib] i2c-2: reading address 0x32, offset 7 failed: No such device or address
11-03 12:07:55.963412 [onlplib] i2c-2: reading address 0x32, offset 9 failed: No such device or address
11-03 12:07:55.970094 [onlplib] i2c-2: reading address 0x32, offset 7 failed: No such device or address
11-03 12:07:55.976763 [onlplib] i2c-2: reading address 0x32, offset 6 failed: No such device or address
11-03 12:07:55.983433 [onlplib] i2c-2: reading address 0x32, offset 8 failed: No such device or address
11-03 12:07:55.990104 [onlplib] i2c-2: reading address 0x32, offset 8 failed: No such device or address
11-03 12:07:55.996773 [onlplib] i2c-2: reading address 0x32, offset 8 failed: No such device or address
11-03 12:07:56.003443 [onlplib] i2c-2: reading address 0x32, offset 9 failed: No such device or address
11-03 12:07:56.010113 [onlplib] i2c-2: reading address 0x32, offset 9 failed: No such device or address
11-03 12:07:56.016782 [onlplib] i2c-2: reading address 0x32, offset 8 failed: No such device or address
11-03 12:07:56.023453 [onlplib] i2c-2: reading address 0x32, offset 8 failed: No such device or address
11-03 12:07:56.030145 [onlplib] i2c-2: reading address 0x32, offset 8 failed: No such device or address
11-03 12:07:56.036822 [onlplib] i2c-2: reading address 0x32, offset 8 failed: No such device or address
11-03 12:07:56.043513 [onlplib] i2c-2: reading address 0x32, offset 3 failed: No such device or address
11-03 12:07:56.043570 [x86_64_delta_ag7648] Unable to read PSU 1 present value)

11-03 12:07:56.050224 [onlplib] i2c-2: reading address 0x32, offset 3 failed: No such device or address
11-03 12:07:56.050282 [x86_64_delta_ag7648] Unable to read PSU 2 present value)

11-03 12:07:56.056924 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.056980 [x86_64_delta_ag7648] Unable to read presence from the port 1 to 9 value(status -15) 

11-03 12:07:56.063613 [onlplib] i2c-2: reading address 0x33, offset 14 failed: No such device or address
11-03 12:07:56.063670 [x86_64_delta_ag7648] Unable to read rx los from the port 1 to 9 value. 

11-03 12:07:56.070294 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.070350 [x86_64_delta_ag7648] Unable to read present status from port(1)

11-03 12:07:56.081905 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.081970 [x86_64_delta_ag7648] Unable to read present status from port(2)

11-03 12:07:56.095026 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.095100 [x86_64_delta_ag7648] Unable to read present status from port(3)

11-03 12:07:56.108146 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.108204 [x86_64_delta_ag7648] Unable to read present status from port(4)

11-03 12:07:56.121266 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.121323 [x86_64_delta_ag7648] Unable to read present status from port(5)

11-03 12:07:56.506085 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.506152 [x86_64_delta_ag7648] Unable to read present status from port(6)

11-03 12:07:56.519484 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.519543 [x86_64_delta_ag7648] Unable to read present status from port(7)

11-03 12:07:56.532814 [onlplib] i2c-2: reading address 0x33, offset 2 failed: No such device or address
11-03 12:07:56.532871 [x86_64_delta_ag7648] Unable to read present status from port(8)

11-03 12:07:56.546156 [onlplib] i2c-2: reading address 0x33, offset 3 failed: No such device or address
11-03 12:07:56.546214 [x86_64_delta_ag7648] Unable to read present status from port(9)

11-03 12:07:56.559482 [onlplib] i2c-2: reading address 0x33, offset 3 failed: No such device or address
11-03 12:07:56.559540 [x86_64_delta_ag7648] Unable to read present status from port(10)

11-03 12:07:56.572802 [onlplib] i2c-2: reading address 0x33, offset 3 failed: No such device or address
11-03 12:07:56.572860 [x86_64_delta_ag7648] Unable to read present status from port(11)

11-03 12:07:56.586194 [onlplib] i2c-2: reading address 0x33, offset 3 failed: No such device or address
11-03 12:07:56.586251 [x86_64_delta_ag7648] Unable to read present status from port(12)

11-03 12:07:56.599574 [onlplib] i2c-2: reading address 0x33, offset 3 failed: No such device or address
11-03 12:07:56.599632 [x86_64_delta_ag7648] Unable to read present status from port(13)

11-03 12:07:56.606264 [onlplib] i2c-2: reading address 0x33, offset 3 failed: No such device or address
11-03 12:07:56.606321 [x86_64_delta_ag7648] Unable to read present status from port(14)

11-03 12:07:56.612953 [onlplib] i2c-2: reading address 0x33, offset 3 failed: No such device or address
11-03 12:07:56.613010 [x86_64_delta_ag7648] Unable to read present status from port(15)

11-03 12:07:56.619632 [onlplib] i2c-2: reading address 0x33, offset 3 failed: No such device or address
11-03 12:07:56.619690 [x86_64_delta_ag7648] Unable to read present status from port(16)

11-03 12:07:56.626314 [onlplib] i2c-2: reading address 0x33, offset 4 failed: No such device or address
11-03 12:07:56.626371 [x86_64_delta_ag7648] Unable to read present status from port(17)

11-03 12:07:56.632995 [onlplib] i2c-2: reading address 0x33, offset 4 failed: No such device or address
11-03 12:07:56.633052 [x86_64_delta_ag7648] Unable to read present status from port(18)

11-03 12:07:56.639684 [onlplib] i2c-2: reading address 0x33, offset 4 failed: No such device or address
11-03 12:07:56.639741 [x86_64_delta_ag7648] Unable to read present status from port(19)

11-03 12:07:56.646364 [onlplib] i2c-2: reading address 0x33, offset 4 failed: No such device or address
11-03 12:07:56.646421 [x86_64_delta_ag7648] Unable to read present status from port(20)

11-03 12:07:56.653046 [onlplib] i2c-2: reading address 0x33, offset 4 failed: No such device or address
11-03 12:07:56.653103 [x86_64_delta_ag7648] Unable to read present status from port(21)

11-03 12:07:56.659734 [onlplib] i2c-2: reading address 0x33, offset 4 failed: No such device or address
11-03 12:07:56.659790 [x86_64_delta_ag7648] Unable to read present status from port(22)

11-03 12:07:56.666414 [onlplib] i2c-2: reading address 0x33, offset 4 failed: No such device or address
11-03 12:07:56.666471 [x86_64_delta_ag7648] Unable to read present status from port(23)

11-03 12:07:56.673094 [onlplib] i2c-2: reading address 0x33, offset 4 failed: No such device or address
11-03 12:07:56.673152 [x86_64_delta_ag7648] Unable to read present status from port(24)

11-03 12:07:56.679784 [onlplib] i2c-2: reading address 0x33, offset 5 failed: No such device or address
11-03 12:07:56.679841 [x86_64_delta_ag7648] Unable to read present status from port(25)

11-03 12:07:56.686484 [onlplib] i2c-2: reading address 0x33, offset 5 failed: No such device or address
11-03 12:07:56.686541 [x86_64_delta_ag7648] Unable to read present status from port(26)

11-03 12:07:56.693165 [onlplib] i2c-2: reading address 0x33, offset 5 failed: No such device or address
11-03 12:07:56.693223 [x86_64_delta_ag7648] Unable to read present status from port(27)

11-03 12:07:56.699864 [onlplib] i2c-2: reading address 0x33, offset 5 failed: No such device or address
11-03 12:07:56.699934 [x86_64_delta_ag7648] Unable to read present status from port(28)

11-03 12:07:56.706562 [onlplib] i2c-2: reading address 0x33, offset 5 failed: No such device or address
11-03 12:07:56.706619 [x86_64_delta_ag7648] Unable to read present status from port(29)

11-03 12:07:56.713243 [onlplib] i2c-2: reading address 0x33, offset 5 failed: No such device or address
11-03 12:07:56.713299 [x86_64_delta_ag7648] Unable to read present status from port(30)

11-03 12:07:56.719925 [onlplib] i2c-2: reading address 0x33, offset 5 failed: No such device or address
11-03 12:07:56.719982 [x86_64_delta_ag7648] Unable to read present status from port(31)

11-03 12:07:56.726614 [onlplib] i2c-2: reading address 0x33, offset 5 failed: No such device or address
11-03 12:07:56.726672 [x86_64_delta_ag7648] Unable to read present status from port(32)

11-03 12:07:56.733305 [onlplib] i2c-2: reading address 0x33, offset 6 failed: No such device or address
11-03 12:07:56.733363 [x86_64_delta_ag7648] Unable to read present status from port(33)

11-03 12:07:56.739995 [onlplib] i2c-2: reading address 0x33, offset 6 failed: No such device or address
11-03 12:07:56.740053 [x86_64_delta_ag7648] Unable to read present status from port(34)

11-03 12:07:57.119434 [onlplib] i2c-2: reading address 0x33, offset 6 failed: No such device or address
11-03 12:07:57.119498 [x86_64_delta_ag7648] Unable to read present status from port(35)

11-03 12:07:57.126126 [onlplib] i2c-2: reading address 0x33, offset 6 failed: No such device or address
11-03 12:07:57.126185 [x86_64_delta_ag7648] Unable to read present status from port(36)

11-03 12:07:57.132814 [onlplib] i2c-2: reading address 0x33, offset 6 failed: No such device or address
11-03 12:07:57.132871 [x86_64_delta_ag7648] Unable to read present status from port(37)

11-03 12:07:57.139514 [onlplib] i2c-2: reading address 0x33, offset 6 failed: No such device or address
11-03 12:07:57.139572 [x86_64_delta_ag7648] Unable to read present status from port(38)

11-03 12:07:57.146206 [onlplib] i2c-2: reading address 0x33, offset 6 failed: No such device or address
11-03 12:07:57.146263 [x86_64_delta_ag7648] Unable to read present status from port(39)

11-03 12:07:57.152894 [onlplib] i2c-2: reading address 0x33, offset 6 failed: No such device or address
11-03 12:07:57.152961 [x86_64_delta_ag7648] Unable to read present status from port(40)

11-03 12:07:57.159583 [onlplib] i2c-2: reading address 0x33, offset 7 failed: No such device or address
11-03 12:07:57.159640 [x86_64_delta_ag7648] Unable to read present status from port(41)

11-03 12:07:57.166263 [onlplib] i2c-2: reading address 0x33, offset 7 failed: No such device or address
11-03 12:07:57.166319 [x86_64_delta_ag7648] Unable to read present status from port(42)

11-03 12:07:57.172946 [onlplib] i2c-2: reading address 0x33, offset 7 failed: No such device or address
11-03 12:07:57.173002 [x86_64_delta_ag7648] Unable to read present status from port(43)

11-03 12:07:57.179634 [onlplib] i2c-2: reading address 0x33, offset 7 failed: No such device or address
11-03 12:07:57.179691 [x86_64_delta_ag7648] Unable to read present status from port(44)

11-03 12:07:57.186324 [onlplib] i2c-2: reading address 0x33, offset 7 failed: No such device or address
11-03 12:07:57.186382 [x86_64_delta_ag7648] Unable to read present status from port(45)

11-03 12:07:57.193006 [onlplib] i2c-2: reading address 0x33, offset 7 failed: No such device or address
11-03 12:07:57.193062 [x86_64_delta_ag7648] Unable to read present status from port(46)

11-03 12:07:57.199694 [onlplib] i2c-2: reading address 0x33, offset 7 failed: No such device or address
11-03 12:07:57.199752 [x86_64_delta_ag7648] Unable to read present status from port(47)

11-03 12:07:57.206384 [onlplib] i2c-2: reading address 0x33, offset 7 failed: No such device or address
11-03 12:07:57.206441 [x86_64_delta_ag7648] Unable to read present status from port(48)

11-03 12:07:57.213076 [onlplib] i2c-2: reading address 0x32, offset 12 failed: No such device or address
11-03 12:07:57.213133 [x86_64_delta_ag7648] Unable to read present status from port(49)

11-03 12:07:57.219764 [onlplib] i2c-2: reading address 0x32, offset 12 failed: No such device or address
11-03 12:07:57.219820 [x86_64_delta_ag7648] Unable to read present status from port(50)

11-03 12:07:57.226444 [onlplib] i2c-2: reading address 0x32, offset 12 failed: No such device or address
11-03 12:07:57.226501 [x86_64_delta_ag7648] Unable to read present status from port(51)

11-03 12:07:57.233136 [onlplib] i2c-2: reading address 0x32, offset 12 failed: No such device or address
11-03 12:07:57.233193 [x86_64_delta_ag7648] Unable to read present status from port(52)

11-03 12:07:57.239824 [onlplib] i2c-2: reading address 0x32, offset 12 failed: No such device or address
11-03 12:07:57.239881 [x86_64_delta_ag7648] Unable to read present status from port(53)

11-03 12:07:57.246525 [onlplib] i2c-2: reading address 0x32, offset 12 failed: No such device or address
11-03 12:07:57.246581 [x86_64_delta_ag7648] Unable to read present status from port(54)
```

A fix similar to https://github.com/bisdn/meta-open-network-linux/pull/70 might be needed, but I'm unsure where and how.

Reply:
doing

```
--- a/packages/platforms/delta/x86-64/ag7648/modules/builds/x86-64-delta-ag7648-i2c-mux-setting.c
+++ b/packages/platforms/delta/x86-64/ag7648/modules/builds/x86-64-delta-ag7648-i2c-mux-setting.c
@@ -517,6 +517,8 @@ static int __init delta_switch_init(void)
 
        pr_debug("DMI Matched %s\n", dmi_id->ident);
 
+       request_module("i2c-i801");
+
        dta_switch = delta_switch_prepare((void *)dmi_id->driver_data);
        if (IS_ERR(dta_switch))
                return PTR_ERR(dta_switch);
```

seems to fix it (and work for both =y and =m), we should definitely delay this until after the next release.

Reply:
Fine with me. Thanks for spotting the problem and looking into it.

Reply:
A fix has been merged in https://github.com/bisdn/meta-open-network-linux/pull/97.